### ISSUE TITLE
Accessibility: Confusing/incorrect navigation bar

### ISSUE THREAD

**Issue by ndarilek** [loconomics/loconomics#638]
I recently restarted my browser, which caused it to launch unmaximized. On my laptop screen, this appears to render a different viewport size where some navigation menus are collapsed, and the nature of these collapses is inaccessible.

 * There are two collapsed buttons whose purpose is unclear.
 * Clicking on them provides no indication that anything has happened, other than to speak that the button is now expanded. Clicking should probably redirect focus to the first item.

I now discover that the first button appears to pop up the main application menu, while the second gives me links to add new items--bookings, appointments, etc.

It seems like you're attempting to use dropdown menus here? If so, can you please direct me to the markup that implements these? At the very least I should be able to get your markup correct. Dropdown menus are pretty finicky if you don't use the right hierarchy, set the correct roles, etc. and it'd probably be easier for me to fix and experiment than to send in various proposed changes.

**Comment by IagoSRL** [comment]
Sorry to forgot this @ndarilek.

Yes, there are collapsed menus, used on narrow viewports (tablet/mobile size, "responsive design"). The narrow navigation bar that contains that is at several files, starting at /app/source/html/parts/AppNav.html, you have the button details at \app\source\html\kocomponents\smart-nav-bar-action.html (there are two root elements there: a link and a button, they switch dynamically and the button is the one responsible to display the collapsed menu), and the dropdown menu contents is at \app\source\html\parts\UserMenuContents.html.

The code around this is more complex than expected, something we have pending of refactor, but the collapsed buttons use just a Bootstrap plugin, maybe we can add some global code to make them accessible (at app.js is fine for now)

**Comment by ndarilek** [comment]
Would you like me to take a crack at this?

If you're just using the stock Bootstrap plugin for this, it should just work out of the box. Are these intended to be dropdowns? Bootstrap dropdowns work very nicely, and I use them heavily myself. If not, could they be refactored to stock Bootstrap dropdowns?

**Comment by joshdanielson** [comment]
@ndarilek I'll let @IagoSRL direct you on this. I'm not sure if he's still traveling (they were going to visit some family after they returned to Spain).

**Comment by IagoSRL** [comment]
@ndarilek We are using Bootstrap ['collapse'](https://getbootstrap.com/docs/3.3/javascript/#collapse) plugin for mobile menus

**Comment by ndarilek** [comment]
Did a bit more digging into this. First, in appNav.html you have:

```
            <h1 id="mainMenuHeaderMobile"class="sr-only App-menus-group-header">MAIN MENU</h1>
```

I'm not used to encountering `<h1/>` in a menu. If it doesn't serve some purpose, I'd recommend removing it. At the very least it's probably confusing the plugin's built-in accessible behavior.

In smart-nav-bar-action.html, you have a `<h1/>` in a `<a/>`, which is also potentially confusing markup to screen readers.

Also in smart-nav-bar-action.html, why do you have both a button and a link to expand the menu? This explains why, when tabbing through the interface, I found both links and buttons that appeared to do the same thing and was significantly confused. :) Were I implementing this, I'd eliminate the link, as semantically a link is expected to transition to a new page while a button is expected to perform an action on the existing page (I.e. uncollapsing the menu.) If there's some reason that both must be present, I'd prefer to add `aria-hidden="true"` to the link.

Assuming nothing is wrong with these proposed changes, I'll be happy to implement them, and I don't immediately see a reason why doing so won't make these menus entirely accessible and no longer confusing to me as a screen reader user. :)

**Comment by ndarilek** [comment]
BTW, I've made all of these proposed changes [here](https://github.com/loconomics/loconomics/tree/issue638). I'm unfortunately blocked by #714 at the moment, and while I could probably bisect to find a working master, I'll wait for the missing code to be committed before testing this out and submitting a PR.

**Comment by IagoSRL** [comment]
@joshdanielson Can you review the 'MAIN MENU' H1?

About the smart-nav-bar-action: well, the link there is in use but only depending on binding values: either the link is in use (navigates to other place) or the button is in use (opens a menu). The 'visible' binding is in use for that (I thought that by hidden and element with "display=none" was not read by screen readers, a bit confused why is a problem here, need to research).
Actually, I'm not very proud of how this piece of code ends behaving (and all the 'smart-nav-bar') and want to refactor it since it causes lot other problems (issue #210). Maybe I can found a quick way to make it behaves accessible now before a full refactor, researching the problem with the use of 'visible'.
The H1 inside the smart-nav-bar-action can be removed, but have to adjust related styles.

**Comment by joshdanielson** [comment]
@IagoSRL it looks like @ndarilek already removed the h1 tag in this branch. It looks fine to me. Let's review again how to use the headers in menus as I still get confused. I think it's really just avoiding h1 tags, which now makes sense to me. I'm posting this link in #210 as a reference.

**Comment by joshdanielson** [comment]
@IagoSRL reviewing this again and do see an h1 still in AppNav under mainMobileNav. I just committed some chagnes. @ndarilek can you test these in this branch? I added role="menu" to the ul lists and gave them an aria-label as well as removed the h1, revised IDs, and added aria-labels

**Comment by ndarilek** [comment]
Oh, so sorry, for some reason I'm no longer getting email notifications from Github so I missed this. I'll look into why that might be today, but are all the changes you'd like tested on master now? I'll test those as well today.

**Comment by IagoSRL** [comment]
@ndarilek Yes, the PR #721 was merged into master so you can test there.

**Comment by ndarilek** [comment]
Ah, I see the changes you needed tested were made by @joshdanielson. I think I examined those last week and they seemed OK, so #721 after both of our changes was good.

**Comment by IagoSRL** [comment]
OK, thanks
